api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -610,6 +610,12 @@ type HostedClusterSpec struct {
 	//
 	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// labels when specified, define what custome labels are added to the hcp pods.
+	// Valid values are those in https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+	//
+	// +optional
```

Comment:
Please document what happens if you change this day 2.
Please include CEL validation for the allowed input here and include test cases for it within TestOnCreateAPIUX